common/src/main/java/com/adaptivescale/rosetta/common/models/BaseModel.java

```diff
@@ -0,0 +1,28 @@
+package com.adaptivescale.rosetta.common.models;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class BaseModel {
```

Comment:
I like this idea of integrating a BaseModel for additional properties.

Reply:
I've renamed the model to `AbstractModel` to better reflect its purpose. Does this change work for you?